addons/account_edi/models/account_edi_document.py

```diff
@@ -172,8 +172,7 @@ def _postprocess_cancel_edi_results(documents, edi_result):
                         'error': move_result.get('error', False),
                         'blocking_level': move_result.get('blocking_level', DEFAULT_BLOCKING_LEVEL) if move_result.get('error') else False,
                     })
-
-            if moves_to_cancel:
+            if moves_to_cancel and all(doc.state == 'cancelled' and doc.blocking_level != 'error' for doc in moves_to_cancel.edi_document_ids):
```

Comment:
@jva-odoo We would at least need a filter on _needs_web_services here.  The case where multiple web services has one with cancelling and one without, I guess would be really corner case, so just that filter should do it.  (Formats as UBL CII, ... create an edi document too, but do not get cancelled normally)